Source/SwiftLintFramework/Helpers/library_wrapper.swift

```diff
@@ -0,0 +1,195 @@
+    xcrunFindPath,
+    /*
+    These search paths are used when `xcode-select -p` points to
+    "Command Line Tools OS X for Xcode", but Xcode.app exists.
+    */
+    applicationsDir?.xcodeDeveloperDir.toolchainDir,
+    applicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeBetaDeveloperDir.toolchainDir
+].compactMap { path in
+    if let fullPath = path?.usrLibDir, FileManager.default.fileExists(atPath: fullPath) {
+        return fullPath
+    }
+    return nil
+})
+
+/// Returns "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment variable
+///
+/// `launch-with-toolchain` sets the toolchain path to the
+/// "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment variable.
+private let xcodeDefaultToolchainOverride = env("XCODE_DEFAULT_TOOLCHAIN_OVERRIDE")
+
+/// Returns "TOOLCHAIN_DIR" environment variable
+///
+/// `Xcode`/`xcodebuild` sets the toolchain path to the
+/// "TOOLCHAIN_DIR" environment variable.
+private let toolchainDir = env("TOOLCHAIN_DIR")
+
+/// Returns toolchain directory that parsed from result of `xcrun -find swift`
+///
+/// This is affected by "DEVELOPER_DIR", "TOOLCHAINS" environment variables.
+private let xcrunFindPath: String? = {
+    let pathOfXcrun = "/usr/bin/xcrun"
+
+    if !FileManager.default.isExecutableFile(atPath: pathOfXcrun) {
+        return nil
+    }
+
+    guard let output = Exec.run(pathOfXcrun, "-find", "swift").string else {
+        return nil
+    }
+
+    var start = output.startIndex
+    var end = output.startIndex
+    var contentsEnd = output.startIndex
+    output.getLineStart(&start, end: &end, contentsEnd: &contentsEnd, for: start..<start)
+    let xcrunFindSwiftPath = String(output[start..<contentsEnd])
+    guard xcrunFindSwiftPath.hasSuffix("/usr/bin/swift") else {
+        return nil
+    }
+    let xcrunFindPath = xcrunFindSwiftPath.appending(pathComponent: "macosx")
+    // Return nil if xcrunFindPath points to "Command Line Tools OS X for Xcode"
+    // because it doesn't contain `sourcekitd.framework`.
+    if xcrunFindPath.hasPrefix("/Library/Developer/CommandLineTools") {
+        return nil
+    }
+    return xcrunFindPath
+}()
+
+private func appDir(mask: FileManager.SearchPathDomainMask) -> String? {
+    return NSSearchPathForDirectoriesInDomains(.applicationDirectory, mask, true).first
+}
+
+private let applicationsDir = appDir(mask: .systemDomainMask)
+
+private let userApplicationsDir = appDir(mask: .userDomainMask)
+
+private extension String {
+    var toolchainDir: String {
+        return appending(pathComponent: "Toolchains/XcodeDefault.xctoolchain/")
+    }
+
+    var xcodeDeveloperDir: String {
+        return appending(pathComponent: "Xcode.app/Contents/Developer")
+    }
+
+    var xcodeBetaDeveloperDir: String {
+        return appending(pathComponent: "Xcode-beta.app/Contents/Developer")
+    }
+
+    var usrLibDir: String {
+        return appending(pathComponent: "/usr/lib/swift")
```

Comment:
`macosx` should be appended at here:
```suggestion
        return appending(pathComponent: "/usr/lib/swift/macosx")
```